exoscale/api/storage.py

```diff
@@ -734,9 +741,17 @@ def __init__(self, key, secret, endpoint=None, zone=None, max_retries=None):
             endpoint_url=self.endpoint,
             aws_access_key_id=key,
             aws_secret_access_key=secret,
-            config=botocore.client.Config(retries={"max_attempts": self.max_retries}),
+            config=botocore.client.Config(
+                user_agent="{} boto3/{} botocore/{}".format(
+                    self.user_agent, boto3.__version__, botocore.__version__
+                ),
+                retries={"max_attempts": self.max_retries},
+            ),
         )
 
+        if trace:
```

Comment:
Would you mind adding a test checking the logger name, since we're modifying it when `trace` is truthy?

Reply:
Just below we do `boto3.set_stream_logger(name="", ...`, so we unset the logger name, don't we? Or maybe I'm mistaking...

Reply:
It's not "unset" *per se*: this empty value means "no filter" (otherwise we don't have a full trace): https://boto3.amazonaws.com/v1/documentation/api/latest/reference/core/boto3.html#boto3.set_stream_logger